packet/conn.go

```diff
@@ -175,9 +187,29 @@ func (c *Conn) copyN(dst io.Writer, src io.Reader, n int64) (written int64, err
 		}
 		buf := c.copyNBuf[:bcap]
 
-		rd, err := io.ReadAtLeast(src, buf, bcap)
+		var rd int
+		rd, err = io.ReadAtLeast(src, buf, bcap)
+
 		n -= int64(rd)
 
+		// if we've read to EOF, and we have compression then advance the sequence number
+		// and reset the compressed reader to continue reading the remaining bytes
+		// in the next compressed packet.
+		if c.Compression != MYSQL_COMPRESS_NONE && rd < bcap &&
```

Comment:
I just read the comment of `ReadAtLeast`, seems for `goErrors.Is(err, io.ErrUnexpectedEOF) || goErrors.Is(err, io.EOF)` we don't need to check `rd < bcap`

Reply:
Yes agree with you after reviewing the `ReadAtLeast` documentation. I've applied the change suggested.